### PR TITLE
Block: Download Counter

### LINKED ISSUE #130: New Block: Download Counter
Sub-task of #129 

192815889-4267f152-b754-404c-a954-8f637bfd43e1

> Something we could explore on this page is a smoother transition between numbers. We could also review the font we chose and use a monospaced one so the numbers stay in the same position as the numbers increase.

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #130, See #129 — This adds a new block for the Download Counter. It live-updates from a new API endpoint every 5 seconds.

The new API endpoint is `/wporg/v1/core-downloads/[branch]/`, and it uses the same SQL request as the current page. On local sites it uses a workaround to avoid calling a table that doesn't exist, that's described in the code.

There's also a new shortcode, for `stable_branch`, which is a different constant (`WP_CORE_STABLE_BRANCH`) than the existing `latest_release` shortcode. This is used in the heading on the Counter page.

⚠️  This branch is built on #131 (`add/download-subpages`), not `trunk`. The other branch enables the new theme on the download subpages so this can be tested.

### Screenshots

Design-wise, there was a comment about using a monospace font to prevent the numbers jumping around - I went ahead and applied that, but we can still try out different styles. I didn't add any animation, so it just updates to the next number.

https://user-images.githubusercontent.com/541093/194117814-133151c3-399d-4652-9a1a-3f289e9cefb0.mp4

Old version data is still accessible with the `?branch=5.9` query parameter (a current feature of the page).

counter-past

The font scales down on small screens.

counter-small

In the editor, the block doesn't live-update, but it does display the latest count.

editor

### How to test the changes in this Pull Request:

This works best on a sandbox, since you can see the real download counts, but it's still possible to test locally.

1. View the Counter page
2. Try passing different branches, ex: `/download/counter/?branch=5.9`
3. Use the block on other pages to test the editor interface — font, color, & spacing should be customizable

### CODE REVIEW

**Comment by ryelle** [comment]
I had the wrong charcoal color background - it's been updated:

Screen Shot 2022-10-05 at 13 33 38

File: source/wp-content/themes/wporg-main-2022/src/download-counter/view.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+				const [ version ] = heading.innerHTML.match( /[0-9]+\.[0-9]/ ) || [];
+				if ( version ) {
+					heading.innerHTML = heading.innerHTML.replace( version, branch );
+				}
+			}
+
+			setInterval( async () => {
+				try {
+					const count = await apiFetch( { path: `/wporg/v1/core-downloads/${ branch }` } );
+					element.innerHTML = count;
```
Comment (StevenDufresne) [inline source/wp-content/themes/wporg-main-2022/src/download-counter/view.js]: `innerHTML` seems unnecessary here, probably a better practice to use a safe sink: `element.textContent`.

**Comment by jasmussen** [comment]
Can we use [IBM Plex Mono](https://fonts.google.com/specimen/IBM+Plex+Mono) for the font? Or, @beafialho in case you have thoughts on a better monospace font?

Edit, perhaps 200:

Screenshot 2022-10-06 at 09 51 46

**Comment by beafialho** [comment]
Yes, @jasmussen! On the lack of an Inter monospace version, I think IBM Plex Mono Extralight works perfectly.

**Comment by ryelle** [comment]
@jasmussen @beafialho Should IBM Plex Mono be used anywhere there's a monospace font? Like these URLs on [the Source Code page](https://user-images.githubusercontent.com/541093/194115130-3cf3d3dc-feeb-4cce-a394-212308c1d793.png)? Or is this just for this one block?

**Comment by jasmussen** [comment]
It's a beautiful font, I'd love to see it used for any and all monospace cases. But I'll defer to Bea!

**Comment by beafialho** [comment]
> @jasmussen @beafialho Should IBM Plex Mono be used anywhere there's a monospace font?

I agree, this can be used everywhere. In the case of these URL's though, the weight used would be `Regular 400`.

**Comment by ryelle** [comment]
Okay, I've created https://github.com/WordPress/wporg-mu-plugins/issues/290 & https://github.com/WordPress/wporg-parent-2021/issues/47 for adding the new font. I'll make sure that gets applied before the page is launched, but I don't think those need to block merging this block. Is there any other design feedback?

**Comment by ryelle** [comment]
Updated to use IBM Plex Mono:

Screen Shot 2022-10-06 at 17 01 18

**Review by StevenDufresne — COMMENTED** [review]
I've tested the code, works as expected.

The JS looks good, the PHP also looks fine although you could loop someone else for a more detailed review.

I'm not sure if there are more design iterations, so I'll leave approval to others.

**Comment by jasmussen** [comment]
Looks good!

**Comment by beafialho** [comment]
@ryelle can we use the Extra Light weight for big text such as counters, etc? Regular should be used for small text.

Captura de ecrã 2022-10-07, às 09 37 51

**Pull request merged**